File: python/info_vaults.py
#!/usr/bin/env python3
"""
Vaults & Delegation Example

Shows how to query vault information and user delegations.

Setup:
    pip install hyperliquid-sdk

Usage:
    export ENDPOINT="https://your-endpoint.hype-mainnet.quiknode.pro/YOUR_TOKEN"
    export USER_ADDRESS="0x..."  # Optional
    python info_vaults.py
"""

import os
from hyperliquid_sdk import Info

ENDPOINT = os.environ.get("ENDPOINT")
USER = os.environ.get("USER_ADDRESS")

if not ENDPOINT:
    print("Set ENDPOINT environment variable")
    exit(1)

info = Info(ENDPOINT)

print("=" * 50)
print("Vaults & Delegation")
print("=" * 50)

# Vault summaries
print("\n1. Vault Summaries:")
vaults = info.vault_summaries()
print(f"   Total: {len(vaults)}")
for v in vaults[:3]:
    print(f"   - {v.get('name', 'N/A')}: TVL ${v.get('tvl', '?')}")

# User delegations
if USER:
    print(f"\n2. Delegations ({USER[:10]}...):")
    delegations = info.delegations(USER)
    if delegations:
        print(f"   {len(delegations)} active")
    else:
        print("   None")
else:
    print("\n(Set USER_ADDRESS for delegation info)")

print("\n" + "=" * 50)
print("Done!")

File: typescript/info_vaults.ts
#!/usr/bin/env npx ts-node
/**
 * Vaults & Delegation Example
 *
 * Shows how to query vault information and user delegations.
 *
 * Setup:
 *     npm install hyperliquid-sdk
 *
 * Usage:
 *     export ENDPOINT="https://your-endpoint.hype-mainnet.quiknode.pro/YOUR_TOKEN"
 *     export USER_ADDRESS="0x..."  # Optional
 *     npx ts-node info_vaults.ts
 */

import { HyperliquidSDK } from 'hyperliquid-sdk';

const ENDPOINT = process.env.ENDPOINT;
const USER = process.env.USER_ADDRESS;

if (!ENDPOINT) {
  console.log("Set ENDPOINT environment variable");
  process.exit(1);
}

async function main() {
  // Single SDK instance — access everything through sdk.info, sdk.core, sdk.evm, etc.
  const sdk = new HyperliquidSDK(ENDPOINT);
  const info = sdk.info;

  console.log("=".repeat(50));
  console.log("Vaults & Delegation");
  console.log("=".repeat(50));

  // Vault summaries
  console.log("\n1. Vault Summaries:");
  const vaults = await info.vaultSummaries() as unknown[];
  console.log(`   Total: ${vaults.length}`);
  for (const v of vaults.slice(0, 3)) {
    const vault = v as Record<string, unknown>;
    console.log(`   - ${vault.name || 'N/A'}: TVL $${vault.tvl || '?'}`);
  }

  // User delegations
  if (USER) {
    console.log(`\n2. Delegations (${USER.slice(0, 10)}...):`);
    const delegations = await info.delegations(USER) as unknown[];
    if (delegations.length) {
      console.log(`   ${delegations.length} active`);
    } else {
      console.log("   None");
    }
  } else {
    console.log("\n(Set USER_ADDRESS for delegation info)");
  }

  console.log("\n" + "=".repeat(50));
  console.log("Done!");
}

main().catch(console.error);
